front/src/modules/simulationResult/components/SpaceTimeChartWrapper/useTrackOccupancy.ts

```diff
@@ -162,35 +196,133 @@ const useTrackOccupancy = ({
     []
   );
 
+  const toOwnerTimetableItemId = (id: TrainId): TimetableItemId => {
+    if (isTrainScheduleId(id) || isPacedTrainId(id)) return id;
+    return extractPacedTrainIdFromOccurrenceId(id);
+  };
```

Comment:
id is a `TrainId`, so it should never be a `PacedTrainId`.
```suggestion
  const toOwnerTimetableItemId = (id: TrainId): TimetableItemId =>
    isTrainScheduleId(id) ? id : extractPacedTrainIdFromOccurrenceId(id);
```